tests/Doctrine/Tests/DBAL/Functional/ConnectionTest.php

```diff
@@ -72,6 +72,18 @@ public function testTransactionNestingBehavior() : void
         }
     }
 
+    public function testTransactionNestingLevelIsResetOnReconnect() : void
+    {
+        $this->connection->beginTransaction();
+        $this->connection->beginTransaction();
+        self::assertEquals(2, $this->connection->getTransactionNestingLevel());
+        $this->connection->close(); // connection is lost
+        $this->connection->beginTransaction(); // should connect, reset nesting level and increase it once
+        self::assertEquals(1, $this->connection->getTransactionNestingLevel());
+        $this->connection->commit();
+        self::assertEquals(0, $this->connection->getTransactionNestingLevel());
```

Comment:
Please use the scenario from the description. The test as it's written now does not reflect the actual issue.

Reply:
I’m not sure that scenario from description is good for test. It just shows one case that may be affected by nesting level issue. Also the result from the description may happen if for example commit works wrong.

I fixed nesting level reset, so i tested the same.

Do you insist on change test?

Reply:
Without that case, it's not clear why we want to reset the transaction nesting level after establishing a connection. The level is an internal implementation detail of the DBAL, it doesn't have to be tested in a functional way but I'd rather have a functional case which relies on it. Even if it's just one case.

Reply:
Ok.
Strange, but Insert statement fails the test now with "Table not found". Seems like table exist only before closing connection in testing environment. I'll look again a little later.

Reply:
Fixed. It was sqlite connection with memory=true. Please review once more